workflow/rules/long_read.smk

```diff
@@ -119,40 +97,50 @@ rule canu_correct:
         "../envs/canu.yaml"
     threads: 16
     shell:
-        "( if [ -d {params.outdir} ]; then rm -Rf {params.outdir}; fi &&"
-        " canu -correct -nanopore {input} -p {wildcards.sample} -d {params.outdir} genomeSize=30k minOverlapLength=10 minReadLength=200"
-        " useGrid=false {params.for_testing}"
-        " corMMapMerSize=10 corOutCoverage=50000 corMinCoverage=0 maxInputCoverage=20000"
-        " corOverlapper=minimap utgOverlapper=minimap obtOverlapper=minimap"
-        " corConcurrency={params.concurrency}"
-        " cormhapConcurrency={params.concurrency} cormhapThreads={params.concurrency}"
-        " cormmapConcurrency={params.concurrency} cormmapThreads={params.concurrency}"
-        " obtmmapConcurrency={params.concurrency} obtmmapThreads={params.concurrency}"
-        " utgmmapConcurrency={params.concurrency} utgmmapThreads={params.concurrency}"
-        " redConcurrency={params.concurrency} redThreads={params.concurrency}"
-        " ovbConcurrency={params.concurrency}"
-        " ovsConcurrency={params.concurrency}"
-        " oeaConcurrency={params.concurrency})"
-        "> {log} 2>&1"
-
-
-# rule medaka_consensus_reference:
-use rule assembly_polishing_ont as medaka_consensus_reference with:
+        """
+        ( if [ -d {params.outdir} ]; then rm -Rf {params.outdir}; fi &&
+        canu -correct -nanopore {input} -p {wildcards.sample} -d {params.outdir} genomeSize=30k minOverlapLength=10 minReadLength=200 \
+        useGrid=false {params.for_testing} \
+        corMMapMerSize=10 corOutCoverage=50000 corMinCoverage=0 maxInputCoverage=20000 \
+        corOverlapper=minimap utgOverlapper=minimap obtOverlapper=minimap \
+        corConcurrency={params.concurrency} \
+        cormhapConcurrency={params.concurrency} cormhapThreads={params.concurrency} \
+        cormmapConcurrency={params.concurrency} cormmapThreads={params.concurrency} \
+        obtmmapConcurrency={params.concurrency} obtmmapThreads={params.concurrency} \
+        utgmmapConcurrency={params.concurrency} utgmmapThreads={params.concurrency} \
+        redConcurrency={params.concurrency} redThreads={params.concurrency} \
+        ovbConcurrency={params.concurrency} \
+        ovsConcurrency={params.concurrency} \
+        oeaConcurrency={params.concurrency})
+        gzip -d {output.corr_gz}
+        2> {log}
+        """
+
+
+rule clip_adbc_corrected:
     input:
-        fasta="results/{date}/corrected/{sample}/{sample}.correctedReads.fasta.gz",
-        reference="resources/genomes/main.fasta",
+        primer=get_artic_primer,
```

Comment:
Can notramp use .bed files instead of the wired python primer file? Then we could just download the primers directly via an URL and wouldn't have to maintain these python files. Moreover, this would ease the way for generalizing the workflow down the road.

Reply:
It can not only use bed, it must! I guess you either didn't notice or I hadn't pushed the changes yet, but I changed the get_artic_primer function to get the primer-bed file (which i put in the resources folder, but downloading would be much better, I agree) instead of this adapter/primer-python file, which one and only use was for porechop and is already deleted. It wouldn't have been compatible with anything else. Porechop I completely evicted from the workflow.

Reply:
I just noticed that the get_artic_primer function somehow had reverted to its previous state with the .py file. Just mended that.